### PR TITLE
Add java setup to gh actions publish job

### PR DESCRIPTION
**Pull request by cgtz**
We need to have the basic java/gradle/repo setup steps in the publishing
job too.

### CODE REVIEW

File: .github/workflows/github-actions.yml [hunk 1]
```diff
@@ -145,6 +145,34 @@ jobs:
...
+            ~/.gradle/wrapper
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+
+      - name: Grant execute permission for gradlew
+        run: chmod +x gradlew
+
       - name: Publish To JFrog artifactory
         if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/master' && env.PUBLISH == 'true'}}
```
Comment (cgtz) [inline .github/workflows/github-actions.yml:177]: hmm, i think i actually have to move this if block to apply to all of these

Comment (cgtz) [reply .github/workflows/github-actions.yml]: GH actions supports if conditions at the job level, so using that: https://github.blog/changelog/2019-10-01-github-actions-new-workflow-syntax-features/

Comment (cgtz) [reply .github/workflows/github-actions.yml]: Turns out that neither `env` or `secrets` is accessible from a job level conditional, so for now we have to duplicate the if statement on each step: https://github.community/t/conditional-job-using-environment-variable/141995

**Review by Arun-LinkedIn — APPROVED** [review]
lgtm!

**Pull request merged**